### PR TITLE
fix(parsers.avro): Handle timestamp format checking correctly

### LINKED ISSUE #13854: Avro input plugin timestamp format check in Init() is buggy
### Relevant telegraf.conf

```toml
n/a
```

### Logs from Telegraf

```text
n/a
```

### System info

Telegraf master branch

### Docker

_No response_

### Steps to reproduce

Put a bad timestamp format, such as "unix_ps", in your config.

### Expected behavior

  It should fail in Init(), because that's not a valid timestamp format.

### Actual behavior

Failure at runtime, with something like:

could not parse \"testdata/supplied_timestamp/message.avro\": could not parse '1664296121000000' to 'unix_ps'

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by athornton**
- [ ] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #13854

Made check for invalid timestamp format happen *outside* the if clause that only fired if the format was the empty string.

### CODE REVIEW

**Comment by athornton** [comment]
We might want to dump this one in favor of #13856, which subsumes it (handling the format correctly turned out to be useful for testing that fix).

File: plugins/parsers/avro/parser.go [hunk 1]
```diff
@@ -57,7 +57,11 @@ func (p *Parser) Init() error {
...
 		}
-		if p.TimestampFormat != "unix" && p.TimestampFormat != "unix_us" && p.TimestampFormat != "unix_ms" && p.TimestampFormat != "unix_ns" {
+	} else {
+		switch p.TimestampFormat {
+		case "unix", "unix_ns", "unix_us", "unix_ms":
+			// Valid values
+		default:
 			return fmt.Errorf("invalid timestamp format '%v'", p.TimestampFormat)
 		}
 	}
```
Comment (srebhan) [inline plugins/parsers/avro/parser.go:67]: How about replacing line 56 to 67 (including) with
```golang
	switch p.TimestampFormat {
	case "":
		p.TimestampFormat = "unix"
	case "unix", "unix_ns", "unix_us", "unix_ms":
		// Valid values
	default:
		return fmt.Errorf("invalid timestamp format '%v'", p.TimestampFormat)
	}
```
as this is what the config has as defaults. You don't use the format if no `Timestamp` is given anyway...

**Review by srebhan — COMMENTED** [review]
Thanks for the fix @athornton! Just one small comment...
I think we should keep this fix in a separate PR as the other PR also changes the behavior of the metric output and that might be controversial...

File: plugins/parsers/avro/parser.go (hunk 1 above)
Comment (athornton) [reply plugins/parsers/avro/parser.go:67]: That does change the behavior of the test case where you don't specify the timestamp format, but probably for the better.

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks @athornton!

**Comment by powersj** [comment]
I am hesitating hitting merge due to the 386 test failure. Do we really need to increase the timeout or is this branch behind master? Or something else?

**Comment by athornton** [comment]
We're not behind master anymore, and the error seems to indicate we really do need longer:

```
INFO [config_reader] Config search paths: [./ /go/src/github.com/influxdata/telegraf /go/src/github.com/influxdata /go/src/github.com /go/src /go / /root] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 33 linters: [asasalint asciicheck bidichk bodyclose depguard dogsled errcheck errname errorlint exportloopref gocheckcompilerdirectives gocritic goprintffuncname gosec gosimple govet ineffassign interfacebloat lll makezero nakedret nilerr nolintlint prealloc predeclared revive sqlclosecheck staticcheck tenv tparallel unconvert unparam unused] 

Too long with no output (exceeded 10m0s): context deadline exceeded
```

**Comment by powersj** [comment]
I find that hard to believe given all you did was modify a switch statement ;)

Looking at master you can see we did not max out the system at all:
https://app.circleci.com/pipelines/github/influxdata/telegraf/17914/workflows/307b4206-1dfd-44cd-94d2-3eaa5e3868a1/jobs/276828/resources

I'm going to land this and hope this is just something strange with the `lsst-sqre` org.

**Comment by athornton** [comment]
It may well be something strange with lsst-sqre, given that we moved away from CircleCI and probably dropped from a paid tier to their free one (we found GitHub Actions to me a much more pleasant experience).

**Pull request merged**